src/select/styled-components.js

```diff
@@ -107,18 +107,27 @@ export const SelectComponentIcon = styled('img', props => {
   }
 });
 
-export const DropDown = styled(MenuList, ({$theme, $isOpen, $type, $rows}) => ({
-  height: $rows ? parseInt($theme.sizing.scale600) * $rows + 'px' : null,
-  overflowY: $rows ? 'scroll' : null,
-  display: !$isOpen ? 'none' : null,
-  top: $type === TYPE.select ? $theme.sizing.scale600 : null,
-  width: `calc(100% - ${$theme.sizing.scale600})`,
-  left: $theme.sizing.scale300,
-  position: 'absolute',
-  listStyle: 'none',
-  borderRadius: $theme.sizing.scale300,
-  boxShadow: $theme.lighting.shadow600,
-}));
+export const DropDown = styled(
+  MenuList,
+  ({$theme, $isOpen, $type, $rows, $itemRef}) => {
+    let maxHeight = null;
+    if ($rows && __BROWSER__ && $itemRef && $itemRef.current) {
+      maxHeight = $itemRef.current.clientHeight * $rows + 'px';
```

Comment:
We shouldn't set a maxHeight based on the height of a single dropdown option.

Reply:
this is actually the main fix itself. So we need to set the height (max height) of the list to the amount of rows, where each row's height is unknown. I have searched possible solution. So far there is no way to do it via css, as this value is dynamic and changing over the rendering cycle. If you have working solution, I can change it accordding to it.